chain/builder.go

```diff
@@ -181,8 +182,15 @@ func (c *Builder) BuildBlock(ctx context.Context, parentView state.View, parent
 		e := executor.New(streamBatch, c.config.TransactionExecutionCores, MaxKeyDependencies, c.metrics.executorBuildRecorder)
 		pending := make(map[ids.ID]*Transaction, streamBatch)
 		var pendingLock sync.Mutex
+		totalTxSize := 0
 		for li, ltx := range txs {
 			txsAttempted++
+			if totalTxSize += ltx.Size(); totalTxSize > consts.MaxTotalTxSizePerBlock {
+				c.log.Debug("Transactions in block exceeded allotted limit ", zap.Int("size", ltx.Size()))
+				restorable = append(restorable, txs[li:]...)
+				break
```

Comment:
We should differentiate between a target during block building and an actual maximum size that is enforced in both block building and block execution/verification.

If this is the maximum size, we should add it into `Execute` as well.